lib/efm32/lg/dma.c

```diff
@@ -0,0 +1,595 @@
+ * @retval true if flag is set
+ * @retval false if flag is not set
+ */
+bool dma_get_request_flag(uint8_t ch)
+{
+	uint32_t mask = DMA_CHREQSTATUS_CHxSREQSTATUS(ch);
+	return (DMA_CHREQSTATUS & mask) != 0;
+}
+
+/**
+ * Get bus error interrupt flag
+ * @retval true if flag is set
+ * @retval false if flag is not set
+ */
+bool dma_get_bus_error_interrupt_flag(void)
+{
+	return (DMA_IF & DMA_IF_ERR) != 0;
+}
+
+/**
+ * Get channel done interrupt flag
+ * @param[in] ch Channel (use DMA_CHANNELx)
+ * @retval true if flag is set
+ * @retval false if flag is not set
+ *
+ */
+bool dma_get_done_interrupt_flag(uint8_t ch)
+{
+	return (DMA_IF & DMA_IF_CHxDONE(ch)) != 0;
+}
+
+/**
+ * Set bus error interrupt flag
+ */
+void dma_set_bus_error_interrupt_flag(void)
+{
+	DMA_IFS = DMA_IFS_ERR;
+}
+
+/**
+ * Set channel done interrupt flag
+ * @param[in] ch Channel (use DMA_CHANNELx)
+ */
+void dma_set_done_interrupt_flag(uint8_t ch)
+{
+	DMA_IFS = DMA_IFS_CHxDONE(ch);
+}
+
+/**
+ * Clear bus error interrupt flag
+ */
+void dma_clear_bus_error_interrupt_flag(void)
+{
+	DMA_IFC = DMA_IFC_ERR;
+}
+
+/**
+ * Clear channel done interrupt flag
+ * @param[in] ch Channel (use DMA_CHANNELx)
+ */
+void dma_clear_done_interrupt_flag(uint8_t ch)
+{
+	DMA_IFC = DMA_IFC_CHxDONE(ch);
+}
+
+/**
+ * Enable bus error interrupt
+ */
+void dma_enable_bus_error_interrupt(void)
+{
+	DMA_IEN |= DMA_IEN_ERR;
+}
+
+/**
+ * Disable bus error interrupt
+ */
+void dma_disable_bus_error_interrupt(void)
+{
+	DMA_IEN &= ~DMA_IEN_ERR;
+}
+
+/**
+ * Enable channel done interrupt
+ * @param[in] ch Channel (use DMA_CHANNELx)
+ */
+void dma_enable_done_interrupt(uint8_t ch)
+{
+	DMA_IEN |= DMA_IEN_CHxDONE(ch);
+}
+
+/**
+ * Disable channel done interrupt
+ * @param[in] ch Channel (use DMA_CHANNELx)
+ */
+void dma_disable_done_interrupt(uint8_t ch)
+{
+	DMA_IEN &= ~DMA_IEN_CHxDONE(ch);
+}
+
+/**
+ * Set channel source
+ * @param[in] ch Channel (use DMA_CHANNELx)
+ * @param[in] source Source (use DMA_CH_CTRL_SOURCESEL_*)
+ */
+void dma_set_source(uint8_t ch, uint32_t source)
+{
+	DMA_CHx_CTRL(ch) = (DMA_CHx_CTRL(ch) & ~DMA_CH_CTRL_SOURCESEL_MASK) | source;
+}
+
+/**
+ * Set channel source signal
+ * @param[in] ch Channel (use DMA_CHANNELx)
+ * @param[in] signal Signal (use DMA_CH_CTRL_SIGSEL_*)
+ */
+void dma_set_signal(uint8_t ch, uint32_t signal)
+{
+	DMA_CHx_CTRL(ch) = (DMA_CHx_CTRL(ch) & ~DMA_CH_CTRL_SIGSEL_MASK) | signal;
+}
+
+/**
+ * Reset channel
+ * @param[in] ch Channel (use DMA_CHANNELx)
+ */
+void dma_channel_reset(uint8_t ch)
+{
+	/* Disable channel */
+	DMA_CHENC = DMA_CHENC_CHxSENC(ch);
+	/* reset channel alternate desc */
+	DMA_CHALTC = DMA_CHALTC_CHxSALTC(ch);
+	/* reset channel priority */
+	DMA_CHPRIC = DMA_CHPRIC_CHxSPRIC(ch);
+	/* clear channel interrupt */
+	DMA_IFC = DMA_IFC_CHxDONE(ch);
+	/* disable loop */
+	if(CHANNEL_SUPPORT_LOOP(ch)) {
+		DMA_LOOPx(ch) = 0;
+	}
+	/* reset signal {source, select} */
+	DMA_CHx_CTRL(ch) = 0;
+}
+
+/**
+ * Set channel loop width
```

Comment:
"set channel loop width", but function is named "dma_set_loop_count"  -> slight inconsistency